### PR TITLE
Fix (Pool) - Added way to LegalAddress to be Main Site Address in Pool

### PR DESCRIPTION
**Pull request by alexsilva-CGI**
## Description

Currently, it is not possible in the Pool for a legal address to be also a main site address at the same time. This PR adjust the way a Address is persisted if both Legal Entity and Site share it.

Related issues: #657 

## Pre-review checks

Please ensure to do as many of the following checks as possible, before asking for committer review:

- [x] DEPENDENCIES are up-to-date. [Dash license tool](https://github.com/eclipse/dash-licenses). Committers can open IP issues for restricted libs.
- [x] [Copyright and license header](https://eclipse-tractusx.github.io/docs/release/trg-7/trg-7-02) are present on all affected files

### CODE REVIEW

**Comment by alexsilva-CGI** [comment]
@nicoprow Tested with the round trip, and it seems that its now associating both site and legalEntity to an address when the LegalAndSiteMainAddress is present on the AddressType. 

I do have a question. Should also this logic apply when an update is done? so for example when type changes from SiteMainAddress to LegalAndSiteMainAddress, an association from both Site and Legal Entity to a single Address is done, with the deletion of the old persisted addresses. Thanks

**Comment by nicoprow** [comment]
> @nicoprow Tested with the round trip, and it seems that its now associating both site and legalEntity to an address when the LegalAndSiteMainAddress is present on the AddressType.
> 
> I do have a question. Should also this logic apply when an update is done? so for example when type changes from SiteMainAddress to LegalAndSiteMainAddress, an association from both Site and Legal Entity to a single Address is done, with the deletion of the old persisted addresses. Thanks

For the update I would follow the same principle we have done before: The parent relationship between L/S/A business partners doesn't change

File: bpdm-pool/src/main/kotlin/org/eclipse/tractusx/bpdm/pool/service/TaskStepBuildService.kt [hunk 1]
```diff
@@ -296,10 +297,19 @@ class TaskStepBuildService(
             val bpnS = bpnIssuingService.issueSiteBpns(1).single()
             val createSite = BusinessPartnerBuildService.createSite(siteDto, bpnS, legalEntity)
             taskEntryBpnMapping.addMapping(bpnSReference, bpnS)
-            val address = createLogisticAddress(mainAddress, taskEntryBpnMapping)
-            createSite.mainAddress = address
-            address.site = createSite
-            createSite
+
+            if (taskEntry.generic.postalAddress.addressType == AddressType.LegalAndSiteMainAddress) { //Association of Site Main Address to Legal Address
```
Comment (nicoprow) [inline bpdm-pool/src/main/kotlin/org/eclipse/tractusx/bpdm/pool/service/TaskStepBuildService.kt:301]: There is some duplicated code here. Why not somthing like:

```
val siteMainAddress = 
            if(genericBusinessPartner.postalAddress.addressType == AddressType.LegalAndSiteMainAddress) 
                legalEntity.legalAddress 
            else  
                createLogisticAddress(mainAddress, taskEntryBpnMapping)
        
  createSite.mainAddress = siteMainAddress
  siteMainAddress.site = createSite
  createSite
```

**Review by nicoprow — APPROVED** [review]

**Review by nicoprow — CHANGES_REQUESTED** [review]
See the comment

**Comment by dilipdhankecha2530** [comment]
@nicoprow @alexsilva-CGI 
We also need to think for the new endpoints which will do the same. In the current development there is no endpoint which we can use to create LegalEntity address as a SiteAddress.

**Pull request merged**